### PR TITLE
Issue1137 Rename variables and add a VAV subpackage

### PR DESCRIPTION
**Pull request by milicag**
This PR addresses #1137:

* Improves the nomenclature within the ```G36_PR1``` package.
* Adds a ```VAV``` suffix to ```SingleZone``` and ```MultiZone``` packages within the ```G36_PR1``` package. 

I'm temporarily committing [the rename script](https://github.com/lbl-srg/modelica-buildings/pull/1139/files#diff-24fe46d23384cb41d3f0d9c235b3618d). Please review and comment on the suggested renames, upon which I can run the script and commit the edits.  Here is my reasoning behind suggested modifications:

1) Even if the model name hints at what the output value refers to (e.g. ```ZoneTemperatures``` output did not contain a ```Zon``` substring), I'd still prefer to spell it out for the purpose of using the same variable name throughout the G36 package. For example ```ZoneTemperatures```: ```THeaSet``` -change to> ```TRooHeaSet```.

2) Any airflow from the terminal unit to the zone should have a ```Dis``` substring.

3) For single zone use ```Roo``` instead of ```Zon```.

4) Use  dT for any differential temperatures.

5) I was conservative to allow long labels in an attempt to minimize ambiguity.

6) Always place ```Set``` at the end of the name, unless the variable is the minimum or the maximum limit, in which case one would use, for example, ```SetMin```.

7) Damper position vs. damper position limit, just an fyi: exhaust damper control signal generating sequences do not have the damper limits and thus generate the damper signal, unlike the rest of the dampers, which generate the damper position, since the physical damper position limits are provided as parameters set at commissioning (as discussed with Brent).

### CODE REVIEW

**Comment by milicag** [comment]
Note that the package renaming script (```move_class``` from ```buildingspy```) did not rename the ```OpenModelica.compareVars``` scripts.

**Comment by milicag** [comment]
@mwetter The PR includes the class rename to add ```VAV```. I'd like some feedback on the variable rename script (link in the PR text) before I run it though.

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+     TCooWinOpe TZonCooSetWinOpe \
+     THeaWinOpe TZonHeaSetWinOpe \     
+
+    # in Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.Reheat.SetPoints.ActiveAirFlow
+
+     VCooMax VDisCooSetMax_flow \
+     VMin VDisSetMin_flow \
+     VHeaMax VDisHeaSetMax_flow \
+     VMinCon VDisConMin_flow \
+     AFlo ZonAre \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: I would keep this `AFlo` (which is also used in the thermal zone model)

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+     outAirPerAre VOut_perAre_flow \
+     outAirPerPer VOut_perPer_flow \
+     VOccMinAir occVDisMin_flow \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: Start with `V`, e.g., `VOccDisMin_flow`

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+     VActMin VActMin_flow \
+     VActHeaMin VActHeaMin_flow \
+     VActHeaMax VActHeaMax_flow \
+
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.Reheat.SystemRequests
+  
+     TCooSet TZonCooSet \
+     VDis VDis_flow \
+     VDisSet VDisSet_flow \
+     TDis \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: Something missing after `TDis`

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+     cooSetDif_1 dTZon_ZonCooSet_1 \ # these are very long
+     cooSetDif_2 dTZon_ZonCooSet_2 \ 
+     disAirSetDif_1 dTDis_DisSet_1 \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: Do we need two times `Dis` in `Dis_Dis`?

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+     TDisSet TDisHeaSet \
+ 
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.ModeAndSetPoints
+    # (most of these need to be consistent with G36_PR1.TerminalUnits.SetPoints.ZoneTemperatures)
+     THeaOn TZonHeaOn \
+     TCooOn TZonCooOn \
+     THeaOff TZonHeaOff \
+     TCooOff TZonCooOff \
+     freProThrVal TZon_frePro_on \
+     freProEndVal TZon_frePro_off \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: Use camel case such as `TZonFreProOn`

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+     TDis \
+     cooSetDif_1 dTZon_ZonCooSet_1 \ # these are very long
+     cooSetDif_2 dTZon_ZonCooSet_2 \ 
+     disAirSetDif_1 dTDis_DisSet_1 \
+     disAirSetDif_2 dTDis_DisSet_2 \ 
+
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.Reheat.DamperValves
+
+     dTDisMax dTDis_ZonSet_Max \
+     V_flow_nominal VNom_flow \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: Keep `V_flow_nominal` as we have `m_flow_nominal` all over `Buildings.Fluid`

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+     decSetDem_1 decTSetDem_1 \
+     decSetDem_2 decTSetDem_2 \
+     decSetDem_3 decTSetDem_3 \
+
+# Buildings.Controls.OBC.ASHRAE.G36_PR1.Generic
+
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.Generic.SetPoints.OperationMode
+     THeaSet occTZonHeaSet \
+     TCooSet occTZonCooSet \
+     TZon TZon \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: TZon TZon are the same

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+# Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.SingleZone
+
+    # SetPoints.ExhaustDamper
+     uFan uSupFan \ 
+
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.SingleZone.SetPoints.OutsideAirFlow
+     AFlo RooAre \ # already occured
+     outAirPerAre VOut_perAre_flow \ # already occured
+     outAirPerPer VOut_perPer_flow \ # already occured
+     TZon TRoo \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: I would always use TZon, for both single and multizone VAV.

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+     AFlo RooAre \ # already occured
+     outAirPerAre VOut_perAre_flow \ # already occured
+     outAirPerPer VOut_perPer_flow \ # already occured
+     TZon TRoo \ 
+     
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.SingleZone.SetPoints.VAVSupply
+     TMax TSupSetMax \
+     TMin TSupSetMin \
+     TSetZon TRooSet \
+     TZon TRoo \
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: Duplicate from above

File: Buildings/issue1137_rename_scripts_temp.txt [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.MultiZone.SetPoints.VAVSupplySignals
+     TSetSup TSupSet \
+     
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.MultiZone.SetPoints.VAVSupplyFan
+     VBox_flow VDis_flow \
+     sumVBox_flow sumVDis_flow \  
+
+    # Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.MultiZone.SetPoints.OutsideAirFlow
+     AFlo ZonAre \ # already occured
```
Comment (mwetter) [inline Buildings/issue1137_rename_scripts_temp.txt]: Keep `AFlo`

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/MultiZoneVAV/Controller.mo [hunk 11]
```diff
@@ -1,4 +1,4 @@
-within Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.MultiZone;
+within Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.MultiZoneVAV;
```
Comment (mwetter) [inline Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/MultiZoneVAV/Controller.mo]: I would prefer `AHUs.VAV.MultiZone`

**Comment by milicag** [comment]
@mwetter I was just about to append to the ```ConvertBuildings_from_5.0_to_6.0.mos```, but the latest versions in the conversion scripts on the master are 4.0 to 5.0. Should I add a new file?

**Comment by mwetter** [comment]
@milicag : Yes, please add a new file.

**Comment by mwetter** [comment]
@milicag : As we already have a package `Buildings.Air.Systems.SingleZone.VAV` we should for consistency call the G36 sequences
`....G36_PR1.AHUs.MultiZone.VAV` and `....G36_PR1.AHUs.SingleZone.VAV`

**Comment by milicag** [comment]
Just in time.

File: Buildings/issue1137_rename_scripts_temp.txt (hunk 4 above)
Comment (milicag) [reply Buildings/issue1137_rename_scripts_temp.txt]: It's simply a T difference between ```TDis``` and ```TDisSet```. Let's ignore that it has the indexes (_1, _2) for now. ```dTDisSet``` is shorter and appealing, but it indicates only one of the variables defining the dT. ```dTDis_Set``` may be a bit too cryptic. Let me know if you have a preference.

Comment (mwetter) [reply Buildings/issue1137_rename_scripts_temp.txt]: If it is a control error, you could call it `errTDis` (= set point minus measured quantity)

Comment (milicag) [reply Buildings/issue1137_rename_scripts_temp.txt]: That fits well, thanks. Do you have any preferences on ```dTDis_ZonSet_Max```? It's the maximum offset.

Comment (mwetter) [reply Buildings/issue1137_rename_scripts_temp.txt]: I would need to know more context but `dTDisZonSetMax` or `errTDis(Max?)` may be options

**Comment by milicag** [comment]
@mwetter I've commited the class rename script as ```Buildings\move_classes_temp.py```. Please let me know if the script runs fine on your environment.

**Comment by mwetter** [comment]
@milicag : I corrected the conversion script and also work on a bug fix in BuildingsPy (https://github.com/lbl-srg/BuildingsPy/pull/194) that caused a new `package.mo` to not be generated. I will need to do some more testing.

**Comment by milicag** [comment]
@mwetter Thank you for creating an "in depth" script. You also modified the conversion script the same way - does that mean that any conversion should be explicit down to the last package in the tree?

**Comment by mwetter** [comment]
@milicag There is no need to list the packages down to the last package in the tree, but
```
convertClass("Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.MultiZone",
                       "Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.MultiZone.VAV")
```
seems ambiguous as it moves `MultiZone` to `MultiZone.VAV`, but then `MultiZone` still exists. Hence, I added one more package level below.

Please try again with the latest version of `BuildingsPy` from the master.

- [ ] Note that `TerminalUnits` should also be moved.

**Comment by milicag** [comment]
@mwetter For debugging purposes, for some reason the script runs the Multizone correctly and throws an error for the single zone:
```
OSError: [Errno 2] No such file or directory: 'Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Economizers'
```
I think there is an issue with building intermediary package. It is also remarkable that the script performs correctly if there is only one downstream layer of packages, but if there are several it apparently fails.

**Comment by mwetter** [comment]
@milicag : I get no error from the script if I apply it to what is on this branch.

**Comment by milicag** [comment]
@mwetter Ok, I just pulled fresh BuildingsPy and am reapplying the script.

**Comment by milicag** [comment]
@mwetter I get that same error. Package order also does not get written.

**Comment by milicag** [comment]
@mwetter That said, I also get a an error when I run the tests:
```
Traceback (most recent call last):
  File "../bin/runUnitTests.py", line 181, in <module>
    show_gui = args.show_gui)
  File "../bin/runUnitTests.py", line 71, in _runUnitTests
    ut = u.Tester(tool=tool)
TypeError: __init__() got an unexpected keyword argument 'tool'
```
Which I have not been able to diagnose, since u.Tester clearly has a ```tool``` kwarg. There may be something odd happening with my setup. Let me attempt the conversion on the laptop.

**Comment by milicag** [comment]
@mwetter I get the same error on the laptop.

**Comment by mwetter** [comment]
@milicag : This error with the unit tests indicate that you have some corrupt installation of BuildingsPy. Try to reinstall it again and make sure you have no old version on your computer.

I also updated the master branch of BuildingsPy so that `package.order` are recreated.

**Comment by milicag** [comment]
@mwetter I just realized it. I'll make sure to have the correct installation.

**Comment by milicag** [comment]
@mwetter ```pip install buildingspy``` is not installing the latest master for some reason. The script and the tests work fine upon installing locally using ```pip install .```. I'll deal with the variables tomorrow, since it's a touchy activity.

**Comment by mwetter** [comment]
@milicag `pip install buildingspy` will install the latest official release from https://pypi.python.org/pypi/buildingspy/1.6.0 and not the master.

**Comment by milicag** [comment]
@mwetter Please note that if tests are run with ```runUnitTests.py``` from master and the latest official release of ```buildingspy```, they will fail due to the missing ```tool``` kwarg. The refactor method ```move_class``` as well. Current ```buildingspy``` master works as expected.

**Comment by milicag** [comment]
@mwetter Would you be able to take a look at the [variable rename script](https://github.com/lbl-srg/modelica-buildings/pull/1139/commits/c531daf564ba3f60c0904ebe63ea398b49b26c6b) and see if I am omitting anything in the methodology. Even if I perform the renames package by package, some of the validation models still fail after the rename. For example, after running the current script (with only a subset of the variables we'd like to rename), the validation model that fails is ```G36_PR1/TerminalUnits/Validation/ModeAndSetPoints```. From what I've seen in the previous variable renaming issues, the script was simply run on the entire library.

**Comment by milicag** [comment]
@mwetter It took me a while. It is not trivial to change a name of a variable, that used to bear the same name in 2 models, into two different names (for the sake of clarity). Please let me know if this is not something I should be attempting to do and I will drop such renames. Example:
```
variable: TCooSet
model: G36_PR1.TerminalUnits.SetPoints.ZoneTemperatures - into TZonCooSet
model: G36_PR1.Generic.SetPoints.OperationMode - into TOccZonCooSet
```
Where the problems occur is when both of these models are used in a, for example model C. In that model the variable name (in this case TCooSet) gets overwritten with say TZonCooSet (for OperationMode connectors) and remains unchanged once the actual OperationMode rename to TOccZonCooSet occurs.

I suspected that if I did:
```
convertElement("Buildings.Controls.OBC.ASHRAE.G36_PR1.Generic.SetPoints.OperationMode", "TZonCooSet ", "TOccZonCooSet")
```
the ```name_of_the_OperationMode_instance.TZonCooSet``` would change to ```name_of_the_OperationMode_instance.TOccZonCooSet```, but the command didn't do it.

How do you usually go about this issue?

**Comment by mwetter** [comment]
I think it makes sense for clarity to use
`G36_PR1.Generic.SetPoints.OperationMode.TOccZonCooSet` (because this model also has `TUnoZonCooSet`)

The `convertElement` function from the `.mos` script should correctly apply the rename in the class as it looks up the class name. But the `.sh` script does not look up the class name, it simply applies the `sed` command, and hence requires manual verification and edits if rename strings are ambiguous.

**Comment by milicag** [comment]
For some reason the convertElement script didn't do anything on my Dymola. Maybe I'm not running it properly. I'll run by Thierry to check for any obvious omissions. 

Regarding the bash script, it may be worth writting a more inteligent variable rename script at some point, that relies on the default component name in order to figure out which strings should and should not be replaced in connectors.

**Comment by milicag** [comment]
@mwetter I'm running into the following issue. The ```convertElement``` command works as expected on a single element, but if I run [an entire convert element script](https://github.com/lbl-srg/modelica-buildings/pull/1139/files#diff-93dee29647d163195f225411f2f25cc8) not all elements get converted. 

The workflow for the variable renames is:
1. The classes have already been renamed in this PR version of the library
2. Run ```convertElement``` script to rename any variables in instantiated classes
3. Run [the sh script](https://github.com/lbl-srg/modelica-buildings/pull/1139/files#diff-9445e06e6649dcd0c729846e67e04a17) to rename the variables in models, classes, scripts and reference results

I've also updated the dymola conversion script. Note that it differs from the script run in step 2 since it contains the class conversions as well, and ```convertElement``` needs to be run on the old paths.

**Comment by milicag** [comment]
There is a need for an another rename:
```.../Validation/Valve.mo``` into ```...Validation/VAVSupplySignals_Valve.mo```, since the previous validation model name did not contain the name of the model validated.

**Comment by milicag** [comment]
I created a new branch for #1137 and intend create a new PR.

**Pull request closed without merging**